### PR TITLE
Remove owner from task json responses

### LINKED ISSUE #11164: api: tasks returns owner in response but should not.
Ownership of resources is to be handled at the /api/v2/{RESOURCE}/{id}/owners endpoint.

For now, tasks needs to remove the field `owner` from the task API.

This is a response that contains owner, but, should not: 

screen shot 2019-01-16 at 12 36 20 pm

/cc @Palakp41 @ebb-tide

### PR DESCRIPTION
**Pull request by lyondhill**
Closes #11164

### CODE REVIEW

File: task/servicetest/servicetest.go [hunk 1]
```diff
@@ -55,7 +55,7 @@ func TestTaskService(t *testing.T, fn BackendComponentFactory) {
 		// but more importantly, it should exercise concurrency to catch data races.
 
 		t.Run("Task CRUD", func(t *testing.T) {
-			t.Parallel()
+			// t.Parallel()
```
Comment (mark-rushakoff) [inline task/servicetest/servicetest.go]: Uncomment?

File: task/servicetest/servicetest.go [hunk 2]
```diff
@@ -110,6 +110,13 @@ type System struct {
...
+	if sys.CredsFunc != nil {
+		var err error
+		orgID, userID, _, err = sys.CredsFunc()
+		if err != nil {
+			t.Fatal(err)
+		}
+	}
 
 	// Create a task.
 	task := &platform.Task{OrganizationID: orgID, Owner: platform.User{ID: userID}, Flux: fmt.Sprintf(scriptFmt, 0)}
```
Comment (mark-rushakoff) [inline task/servicetest/servicetest.go]: I don't necessarily see the path forward here, with omitting the owner from the response but still accepting it in the create arguments.

I understand that https://github.com/influxdata/influxdb/issues/11427 is to track the big picture, and if work needs to be pushed there, that's fine.

Should this test be updated to check the owner/member API to ensure that the owner matches what was provided? Or would that assertion be part of https://github.com/influxdata/influxdb/issues/11427?

Comment (lyondhill) [reply task/servicetest/servicetest.go]: You're right. I think that all TaskCreates should only be able to accept the owner through the context and not in the task struct. It is strange to me that you can set the owner of a task as someone other than yourself.

File: task/servicetest/servicetest.go (hunk 1 above)
Comment (lyondhill) [reply task/servicetest/servicetest.go]: Your right. We should keep this parallel. The problem is we are using the same "system" for all the tests when we should generate a new system for each test.

Comment (mark-rushakoff) [reply task/servicetest/servicetest.go]: I think the reason for using the same system, and t.Parallel, was to exercise that the task systems work correctly under concurrent operation. I would prefer to rearrange the test setup over duplicating systems.

Comment (lyondhill) [reply task/servicetest/servicetest.go]: We wrote a test just for the purpose of testing concurrency. `testTaskConcurrency` It is entirely unnecessary to use the crud test to check the concurrency as well.

Comment (mark-rushakoff) [reply task/servicetest/servicetest.go]: `testTaskConcurrency` should be renamed, as it doesn't exercise all the  methods covered in the other set of tests.

File: task/servicetest/servicetest.go (hunk 2 above)
Comment (lyondhill) [reply task/servicetest/servicetest.go]: After our conversation I thing this is going to be removed in a later pr and is left here until we can make other changes in the backend.

**Pull request merged**